### ISSUE TITLE
3 bogus fields being added to some CSV file headers

### ISSUE THREAD

**Issue by stsccfr** [FreeUKGen/MyopicVicar#2760]
I have found many CSV files in the database that contain 3 invalid extended CSV fields.  These fields are sometimes being added by the FreeREG server to those supplied by the transcriber.  The invalid fields are 'county', 'place', and 'record_type', and all are redundant, because we are already using 'chapman_code' and 'place_name' and the record type is the event type and is specified in the first line of the header.  These 3 fields are not always being added, however, which suggests that it was not intentional to do so, but because they are not part of the extended CSV specification, their presence will cause problems for FreeComETT if we use that program to make corrections to the files.  They are defined in the Free1CsvFile class, where a comment seems to acknowledge that they are redundant, but they are not defined in lib/freereg_options_constants.rb

An example is STSKISBU.csv in the live database (user JenniferLawson). The 6th line of the header containing the FlexCSV field names, and the first record, are as follows:

```
chapman_code,place_name,church_name,register_type,register_entry_number,death_date,burial_person_forename,burial_person_surname,burial_person_title,burial_person_abode,burial_date,person_age,person_title,notes,notes_from_transcriber,county,place,record_type
STS,Kingstone,St John the Baptist,PR,1,"",Richard,BENTLEY,"",Heatley,1 Feb 1813,19,"","","",STS,Kingstone,bu
```

As we see, the 3 redundant fields have been added to the list of fields in the header and to the end of each record in the file.

I am not sure where in the code these 3 extra fields are being added.  It might be when we do `setup_batch_for_processing()` in ./lib/new_freereg_csv_update_processor.rb, but even if that is true, there is still the question of why they aren't *always* in downloaded files.  Are we only adding them sometimes, or are we always adding them but then sometimes removing them when the file is downloaded?  We will need to fix this for FreeComETT.

**Comment by stsccfr** [comment]
@vino I can reproduce this now, at least in the live database (I can't use test3 because the location of the files has moved; what I see in the online interface no longer corresponds to what I see on night).  What you have to do is edit a CSV file containing the "#,DEF" flag using the online editor.  Make some change to one of the record fields, and then download the file to trigger the update.  Then look at the new copy of the file that is in the <userid> directory and it will now have the 3 bogus fields following the "#,DEF" line.  This explains why we see this problem only occasionally.  It isn't happening when uploading a new file, but only during the processing after using the online editor.

**Comment by Vino-S** [comment]
Ready for testing on test3

**Comment by stsccfr** [comment]
The processor is behaving correctly for marriages and burials now, but not for baptisms.  For baptisms the processor is adding the `private_baptism` field to the header and then populating it with `false` for all records.  Attached is a file that can be used for testing.

[STSWOIBA3.CSV](https://github.com/user-attachments/files/20398996/STSWOIBA3.CSV)

Use the online editor to change the surname in the first record from SMITH to SMITHE and then save.  Download the file to trigger the update and then look at the new copy of STSWOIBA3.CSV in your user directory and compare with the `.attic` copy.

**Comment by Vino-S** [comment]
@stsccfr Applied the fix for Baptism as well. Could you test it, please

**Comment by stsccfr** [comment]
Yes, the fix works for baptisms now.  Marking this as Ready for Deployment.

**Comment by DeniseColbert** [comment]
Done: closing

**Comment by stsccfr** [comment]
After dicussion with Vino it appears that the code changes in this case are not responsible for the problems, so I can close this case again.
